client_cli/src/main.rs

```diff
@@ -1023,3 +1118,80 @@ mod json {
         }
     }
 }
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use serde_json::Value as JsonValue;
+
+    use super::*;
+
+    #[test]
+    fn parse_ipv4_address() {
+        let actual = ValueArg::from_str("192.168.0.1").expect("Failed to parse valid IPv4 address");
+        assert_eq!(
+            actual,
+            ValueArg(Value::Ipv4Addr(Ipv4Addr::new([192, 168, 0, 1])))
+        );
+    }
+
+    #[test]
+    fn parse_ipv6_address() {
+        let actual = ValueArg::from_str("::1").expect("Failed to parse valid IPv6 address");
+        assert_eq!(
+            actual,
+            ValueArg(Value::Ipv6Addr(Ipv6Addr::new([0, 0, 0, 0, 0, 0, 0, 1])))
+        );
+    }
+
+    #[test]
+    fn parse_boolean_true() {
+        let actual = ValueArg::from_str("true").expect("Failed to parse 'true' as boolean");
+        assert!(matches!(actual, ValueArg(Value::Bool(true))));
+    }
+
+    #[test]
+    fn parse_boolean_false() {
+        let actual = ValueArg::from_str("false").expect("Failed to parse 'false' as boolean");
+        assert!(matches!(actual, ValueArg(Value::Bool(false))));
+    }
+
+    #[test]
+    fn parse_numeric_value() {
+        let actual = ValueArg::from_str("123_u32").expect("Failed to parse valid numeric value");
+        assert!(matches!(
+            actual,
+            ValueArg(Value::Numeric(NumericValue::U32(123)))
+        ));
+    }
+
+    #[test]
+    fn parse_public_key() {
+        let public_key_str =
+            "ed01207233BFC89DCBD68C19FDE6CE6158225298EC1131B6A130D1AEB454C1AB5183C0";
+        let actual = ValueArg::from_str(public_key_str).expect("Failed to parse valid public key");
+        assert_eq!(
+            actual,
+            ValueArg(Value::PublicKey(
+                PublicKey::from_str(public_key_str).unwrap()
+            ))
+        );
+    }
+
+    #[test]
+    fn parse_json_value() {
+        let json_str = r#"{"Vec":[{"String":"a"},{"String":"b"}]}"#;
```

Comment:
You might also consider using `serde_json::json!` macro:

```rs
json!({"Vec": [{"String": "a"}, {"String": "b"}]})
```

Reply:
I think we should leave it as it is to be able to parse it using from_str we implemented. Using this aproach we would have to parse the object into str after we create it then test from_str.